pkg/controllers/gitrepository/controller.go

```diff
@@ -0,0 +1,280 @@
+package gitrepository
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/cnoe-io/idpbuilder/api/v1alpha1"
+	"github.com/cnoe-io/idpbuilder/globals"
+	"github.com/cnoe-io/idpbuilder/pkg/controllers/localbuild"
+	"github.com/cnoe-io/idpbuilder/pkg/util"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	DefaultBranchName = "main"
+	// DefaultGiteaUsername and DefaultGiteaPassword taken from gitea helm chart. https://gitea.com/gitea/helm-chart#gitea
+	DefaultGiteaUsername = "gitea_admin"
+	DefaultGiteaPassword = "r8sA8CPHD9!bt6d"
```

Comment:
I exposed the secret name for gitea under `resource.Status.GiteaSecretName` for the local build. let's use that instead. The value is based on what we have pushed to the registry here.

Also if we want to make the username and password fancier, we can certainly do it. I dont see particularly a need though. Gitea is set to only accept local connections anyway.

https://github.com/cnoe-io/idpbuilder/blob/11a9914536cbd8ee8da9f5d79e8924f1e595ec0c/pkg/controllers/localbuild/resources/gitea/k8s/install.yaml#L583-L590C4